lineflow/core.py

```diff
@@ -206,6 +206,12 @@ def get_example(self, i: int) -> Any:
     def __len__(self) -> int:
         return super(IterableDataset, self).__len__()
 
+    def __getitem__(self, index: Union[int, slice]) -> Union[Any, List[Any]]:
+        out = super(IterableDataset, self).__getitem__(index)
+        if isinstance(out, list):
+            out = IterableDataset(out)
```

Comment:
It would be great if you could use `Dataset` instead of `IterableDataset`.
`Dataset` is supposed to be used for indexed data structures and `IterableDataset` for non-indexed ones.

Reply:
Thanks, should be fixed now.